lib/rules/id-blacklist.js

```diff
@@ -34,7 +34,7 @@ module.exports = {
         // Helpers
         //--------------------------------------------------------------------------
 
-        var blacklist = context.options;
+        var blacklistRegex = new RegExp("^(" + context.options.join("|") + ")$");
```

Comment:
This could be made easier if we abandon the variable-length option list for regexes.

Config options:

1. Old-style, assumed to be string literals/no regexes: `"id-blacklist: [2, "foo"]`
1. New-style, allows for strings and patterns: `"id-blacklist: [2, { "strings": ["foo"], "patterns": ["^someRegexWithMetacharacters$"] }]`

In that case, old-style config's identifiers, and new-style config's "strings", would be anchored and escaped (if interpreted as regular expressions). New-style config "patterns" would be unmodified and regular expression characters would be allowed.